<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MF_trailerDecoderDialog</class>
 <widget class="QDialog" name="MF_trailerDecoderDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>534</width>
    <height>507</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Trailer Decoder</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_7">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QGroupBox" name="sizeGroupBox">
       <property name="title">
        <string>Blocks</string>
       </property>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QRadioButton" name="size4Button">
          <property name="text">
           <string>4</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QRadioButton" name="size16Button">
          <property name="text">
           <string>16</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Trailer Data:
(like &quot;FF0780&quot; or &quot;FF 07 80&quot;)</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLineEdit" name="accessBitsEdit"/>
       </item>
       <item>
        <widget class="QLabel" name="isAccessBitsValidLabel">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>40</height>
          </size>
         </property>
         <property name="text">
          <string/>
         </property>
         <property name="alignment">
          <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::MinimumExpanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Or set bits manually</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Cx0</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="C0Box">
             <property name="maximum">
              <number>7</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Cx1</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="C1Box">
             <property name="maximum">
              <number>7</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_4">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Cx2</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="C2Box">
             <property name="maximum">
              <number>7</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <property name="spacing">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Cx3</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="C3Box">
             <property name="maximum">
              <number>7</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::MinimumExpanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Data Block Permission:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="dataBlockWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>4</verstretch>
      </sizepolicy>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <row>
      <property name="text">
       <string>Block0</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Block1</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Block2</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>Read</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Write</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Increase</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Decrease/Transfer/Restore</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Trailer Block Permission:</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QTableWidget" name="trailerBlockWidget">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>3</verstretch>
      </sizepolicy>
     </property>
     <property name="editTriggers">
      <set>QAbstractItemView::NoEditTriggers</set>
     </property>
     <row>
      <property name="text">
       <string>Read</string>
      </property>
     </row>
     <row>
      <property name="text">
       <string>Write</string>
      </property>
     </row>
     <column>
      <property name="text">
       <string>KeyA</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>Access Bits</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>KeyB</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Reference:
MF1S70YYX_V1 Product data sheet
Rev. 3.2 â€” 23 November 2017</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDialogButtonBox" name="buttonBox">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="standardButtons">
        <set>QDialogButtonBox::Ok</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>Note:the Access Bits usually contains 4 bytes(8 hex symbols), but only the first 3 bytes matters. You can set the 4th byte randomly.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>MF_trailerDecoderDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>MF_trailerDecoderDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
